test/envtest/konnect_entities_suite_test.go

```diff
@@ -111,3 +113,77 @@ func testNewKonnectEntityReconciler[
 		}
 	})
 }
+
+const (
+	// testCpID is a test Konnect Control Plane ID.
+	testCpID = "cp-id"
+)
+
+// prepareValidKonnectCP creates a KonnectGatewayControlPlane with a reference to a KonnectAPIAuthConfiguration with
+// statuses that make them both usable in tests that require a reference to a valid Konnect Control Plane.
+func prepareValidKonnectCP(ctx context.Context, t *testing.T, ns string, cl client.Client) (cpName string) {
+	auth := &konnectv1alpha1.KonnectAPIAuthConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      uuid.NewString(),
+			Namespace: ns,
+		},
+		Spec: konnectv1alpha1.KonnectAPIAuthConfigurationSpec{
+			Type:      konnectv1alpha1.KonnectAPIAuthTypeToken,
+			Token:     "kpat_test",
+			ServerURL: "127.0.0.1",
+		},
+	}
+	require.NoError(t, cl.Create(ctx, auth))
```

Comment:
Can we use `deployKonnectAPIAuthConfiguration` for this to not duplicate code that does pretty much the same thing?

Feel free to change it to suit the needs here.

Similarly we can use (and possibly change) `deployKonnectAPIAuthConfigurationWithProgrammed` for below status update.